pkg/monitor/monitor/monitor_manager.go

```diff
@@ -427,3 +395,83 @@ func (m *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef,
 	_, err = m.deps.Clientset.PingcapV1alpha1().TidbClusters(tc.Namespace).Patch(tc.Name, types.MergePatchType, mergePatch)
 	return err
 }
+
+func (m *MonitorManager) smoothMigrationToStatefulSet(monitor *v1alpha1.TidbMonitor) (bool, error) {
+	// determine whether there is an old deployment
+	deploymentName := GetMonitorObjectName(monitor)
+	exist, err := m.deps.TypedControl.Exist(client.ObjectKey{
+		Namespace: monitor.Namespace,
+		Name:      deploymentName,
+	}, &appsv1.Deployment{})
+	if err != nil {
+		klog.Errorf("smoothMigration tm[%s/%s]'s,old deployment failed to get,err: %v", monitor.Namespace, monitor.Name, err)
+		return false, err
+	}
+	if exist {
+		klog.Infof("smoothMigration tm[%s/%s]'s,old deployment is exist", monitor.Namespace, monitor.Name)
+		// if deployment exist, delete it and wait next reconcile.
+		err = m.deps.TypedControl.Delete(monitor, &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      deploymentName,
+				Namespace: monitor.Namespace,
+			},
+		})
+		if err != nil {
+			klog.Errorf("smoothMigration tm[%s/%s]'s,old deployment failed to delete,err: %v", monitor.Namespace, monitor.Name, err)
+			return false, err
+		}
+
+		return !monitor.Spec.Persistent, nil
+	} else {
```

Comment:
The following block should also be done if the deployment is found in the above `if` branch.
Why separate it with the deployment deletion logic?